desktop/createDownloadQueue.ts

```diff
@@ -0,0 +1,85 @@
+import type {BrowserWindow} from 'electron';
+import createQueue from '@libs/Queue/Queue';
+import ELECTRON_EVENTS from './ELECTRON_EVENTS';
+import electronDownload from './electronDownloadManager';
+import type {Options} from './electronDownloadManagerType';
+
+type DownloadItem = {
+    // The window where the download will be initiated
+    win: BrowserWindow;
+
+    // The URL of the file to be downloaded
+    url: string;
+
+    // The options for the download, such as save path, file name, etc.
+    options: Options;
+};
+
+/**
+ * Creates a download queue.
+ * @returns An object with methods to enqueue and dequeue download items from the queue.
+ */
+const createDownloadQueue = () => {
+    const queue = createQueue<DownloadItem>();
+
+    /**
+     * Downloads the specified item.
+     * @param item - The item to be downloaded.
+     */
+    const downloadItem = (item: DownloadItem): void => {
+        const newItem = {
+            ...item,
+            options: {
+                ...item.options,
+                onStarted: () => {
+                    item.win.webContents.send(ELECTRON_EVENTS.DOWNLOAD_STARTED, {url: item.url});
+                },
+                onCompleted: () => {
+                    queue.dequeue();
+                    if (queue.isEmpty()) {
+                        return;
+                    }
+
+                    const nextItem = queue.peek();
+                    if (nextItem !== undefined) {
+                        downloadItem(nextItem);
+                    }
```

Comment:
The logic for running / getting next item from the queue should not reside here. It should be in `import createQueue from '@libs/Queue/Queue';`

Reply:
It has been refactored